common/config/archival.go

```diff
@@ -28,28 +28,17 @@ import (
 
 // Validate validates the archival config
 func (a *Archival) Validate(domainDefaults *ArchivalDomainDefaults) error {
-	if !isArchivalConfigValid(a.History.Status, a.History.EnableRead, domainDefaults.History.Status, domainDefaults.History.URI, a.History.Provider != nil) {
-		return errors.New("Invalid history archival config")
+	if a.History.Status == common.ArchivalEnabled {
```

Comment:
If archival is disabled, I think we still need to check if "read" is disabled or not. Since read also requires the archiver to be enabled and initialized. 
For other two for domain default status and URI, I agree that we don't have to check them when archival is disabled.

Reply:
Okay, will add it back.